velero-plugin-for-aws/object_store.go

```diff
@@ -196,6 +209,9 @@ func (o *ObjectStore) Init(config map[string]string) error {
 		if len(caCert) > 0 {
 			sessionOptions.CustomCABundle = strings.NewReader(caCert)
 		}
+		if credentialsFile != "" {
```

Comment:
If this is `!= nil`, does it not need the processing as in this line: https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/69/files#diff-fdece575520884a3b21814a0f7d1d42953950bed4aff84fe9f9ca635b958d717R178

?

Reply:
I skipped it here because if it was not empty, it would already have been checked above but it would probably be better to have the check in both places. I could move the above validation into a separate function and call it in both locations. Would that work for you?

Reply:
Yes, yes, I see. Something threw me off and I think it's this:

LN 184 already sets `sessionOptions.SharedConfigFiles = []string{credentialsFile}` so this is not needed? If there was an error that didn't allow for the `sessionOptions.SharedConfigFiles` to be set, the code would've returned. I don't this LNs 212-214 are needed.

Reply:
Ah, it's because the `sessionOptions` variable is shadowed in this block on line 208 so this is a new object that we're changing here. I wonder if we can use the same `sessionOptions` object for both so that we don't need to performa all these checks again. I'll take a look and try to improve it. I haven't reviewed it thoroughly yet, but this may already be addressed in #71.

Reply:
Oh right, I missed that.

If you change LN 208 from:

`sessionOptions := session.Options{Config: *publicConfig, Profile: credentialProfile}`

to:

`sessionOptions = session.Options{Config: *publicConfig, Profile: credentialProfile}`

would that not just overwrite that struct and allow you to continue using the same instance of `sessionOptions`?

Reply:
@carlisia Yes, it would override it and use the same instance, but I would still need to add the CACert and credentials to it. I added a new function `newSessionOptions` to handle creation of the session so that we use the same logic for both the main session and the public session, which I think it is a bit clearer.